src/service/data_products/genome_attributes.py

```diff
@@ -222,18 +228,51 @@ def _remove_keys(doc):
 _FLD_LIMIT = "limit"
 
 
-# At some point we're going to want to filter/sort on fields. We may want a list of fields
-# somewhere to check input fields are ok... but really we could just fetch the first document
-# in the collection and check the fields 
+@_ROUTER.get(
+    "/meta",
+    response_model=col_models.ColumnarAttributesMeta,
+    description=
+"""
+Get metadata about the genome attributes table including column names, type,
+minimum and maximum values, etc.
+""")
+async def get_genome_attributes_meta(
+    r: Request,
+    collection_id: str = PATH_VALIDATOR_COLLECTION_ID,
+    load_ver_override: common_models.QUERY_VALIDATOR_LOAD_VERSION_OVERRIDE = None,
+    user: kb_auth.KBaseUser = Depends(_OPT_AUTH)
+    ):
+    storage = app_state.get_app_state(r).arangostorage
+    _, load_ver = await get_load_version(storage, collection_id, ID, load_ver_override, user)
+    doc = await get_collection_singleton_from_db(
+            storage,
+            names.COLL_GENOME_ATTRIBS_META,
+            collection_id,
+            load_ver,
+            bool(load_ver_override)
+    )
+    doc[col_models.FIELD_COLUMNS] = [col_models.AttributesColumn.model_construct(**d)
```

Comment:
Is this always ordered (ie is this list always in the same order as the columns in the table?)

Reply:
I'm kinda confused on if this func returns a dict or a list, I think I need to learn more about the pydantic lib to parse this PR

Reply:
`model_construct` constructs an instance of the model class upon which it's called, in this case `AttributesColumn`

Reply:
I know `doc` return by get_collection_singleton_from_db is a dict. Can you provide a sample doc? Would like to know more what keys/values are inside.

Reply:
No samples docs exist currently; doc creation will need to be added to the pipeline parser. Also the contents of the doc will be different depending on the arango collection specified in the arguments. For this case, though, you can see what fields the doc will need to have by examining the model that will be constructed: https://github.com/kbase/collections/blob/main/src/common/product_models/columnar_attribs_common_models.py

Reply:
doc["columns"] will be a list of col names or dict?

Reply:
https://github.com/kbase/collections/blob/main/src/common/product_models/columnar_attribs_common_models.py#L106

Reply:
The part I am trying to understand is L254-255. Maybe I confuse myself.

You already had a list of AttributesColumn and why you construct them again and reassign to doc["columns"]. M

Reply:
What's coming back from the database is just dicts and lists. The response model defined for this method is a `ColumnarAttributesMeta`, and so L254-255 just construct that response model from the dict/list data structure. L254 is necessary because `model_construct` is not recursive, so any enclosed models need to be explicitly created.